### PR TITLE
Add support for Disk Scraper Metrics

### PR DESCRIPTION
**Pull request by ishleenk17**
Adds metrics for the Disk Scraper. 

**Link to tracking Issue:**
https://github.com/elastic/opentelemetry-dev/issues/181

### CODE REVIEW

File: processor/elasticprocessor/internal/hostmetrics/disk.go [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+				} else {
+					continue
+				}
+
+				if direction, ok := dp.Attributes().Get("direction"); ok {
+					name := metric.Name()
+					timestamp := dp.Timestamp()
+					value := dp.IntValue()
+
+					switch direction.Str() {
```
Comment (tommyers-elastic) [inline processor/elasticprocessor/internal/hostmetrics/disk.go]: L26-34 can be simplified to just `addDiskIntMetric(metrics, resource, dataset, metric.Name(), device, direction.Str(), dp.Timestamp(), dp.IntValue())`

File: processor/elasticprocessor/internal/hostmetrics/disk.go [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+	}
+}
+
+// Adds the translated metrics with Double Values
+func addDiskDoubleMetric(metrics pmetric.MetricSlice, resource pcommon.Resource,
+	dataset, name, device, esmetricname string, timestamp pcommon.Timestamp, value float64, multiplier float64) {
+
+	metricsToAdd := map[string]string{
+		"system.disk.operation_time": "system.diskio.%s.time",
+		"system.disk.io_time":        "system.disk.%s.time",
```
Comment (tommyers-elastic) [inline processor/elasticprocessor/internal/hostmetrics/disk.go]: is this meant to be `diskio`?

File: processor/elasticprocessor/internal/hostmetrics/disk.go [hunk 3]
```diff
@@ -0,0 +1,152 @@
...
+				if d, ok := dp.Attributes().Get("device"); ok {
+					device = d.Str()
+				} else {
+					continue
+				}
+				if direction, ok := dp.Attributes().Get("direction"); ok {
+					name := metric.Name()
+					timestamp := dp.Timestamp()
+					value := dp.DoubleValue()
+					if name == "system.disk.operation_time" {
```
Comment (tommyers-elastic) [inline processor/elasticprocessor/internal/hostmetrics/disk.go]: this `if` is redundant

File: processor/elasticprocessor/internal/hostmetrics/disk.go (hunk 1 above)
Comment (ishleenk17) [reply processor/elasticprocessor/internal/hostmetrics/disk.go]: Done!

File: processor/elasticprocessor/internal/hostmetrics/disk.go (hunk 2 above)
Comment (ishleenk17) [reply processor/elasticprocessor/internal/hostmetrics/disk.go]: Yes, thanks

File: processor/elasticprocessor/internal/hostmetrics/disk.go (hunk 3 above)
Comment (ishleenk17) [reply processor/elasticprocessor/internal/hostmetrics/disk.go]: To be future-ready in case we get a new metric ?

Comment (tommyers-elastic) [reply processor/elasticprocessor/internal/hostmetrics/disk.go]: that suggests a code smell to me. there are a couple of others in this file which make me think this code could do with some simplification - for example, the fact we have a parameter which in some cases is the 'direction', and in others a static string `io`. in addition we have two functions with very similar names `addDiskDoubleMetric` and `addDiskIntMetric`, which i might suggest we combine using type parameters, but then when you look deeper, there is non-type-specific logic in there (like the multiplier).

**Comment by ishleenk17** [comment]
The code for disk scraper is now present [here](https://github.com/elastic/opentelemetry-lib/pull/17).

**Pull request closed without merging**